### PR TITLE
Support for push/pull translation in the sink API

### PR DESCRIPTION
**Pull request by djspiewak**
Depends on #4609

I went back and forth on *where* to put this type (e.g. could put it on the package object), but I decided to just not overcomplicate things. The only thing I'm still mulling is whether or not to wrap this up in a more general "SDBE Capabilities" concept, since technically this is one aspect of the server environment that we're exposing as a capability to sinks.

### CODE REVIEW

**Comment by djspiewak** [comment]
You know, I really don't understand Zinc sometimes…

**Comment by djspiewak** [comment]
Published as `169.0.0-b0a7528`

File: connector/src/main/scala/quasar/connector/destination/package.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package quasar.connector
+
+import slamdata.Predef._
+
+import fs2.Pipe
+
+import java.net.URI
+
+package object destination {
+  type PushmiPullyu[F[_]] = (URI => F[Unit]) => Pipe[F, Byte, Unit]
```
Comment (wemrysi) [inline connector/src/main/scala/quasar/connector/destination/package.scala:26]: Seems like a good spot :+1:

File: connector/src/main/scala/quasar/connector/destination/ResultSink.scala [hunk 2]
```diff
@@ -33,17 +33,27 @@ import skolems.∀
...
 object ResultSink {
+
   final case class CreateSink[F[_], T](
       renderConfig: RenderConfig,
-      consume: (ResourcePath, NonEmptyList[Column[T]], Stream[F, Byte]) => Stream[F, Unit])
+      consume: CreateSink.Args[F, T] => Stream[F, Unit])
       extends ResultSink[F, T]
 
+  object CreateSink {
+    final case class Args[F[_], T](
```
Comment (wemrysi) [inline connector/src/main/scala/quasar/connector/destination/ResultSink.scala:43]: :+1:

**Comment by djspiewak** [comment]
> I think we need some docs around this type, not sure whether on the type itself or on ResultSink. Describing the general purpose and especially what the expectations of the URI => F[Unit] function are as the types don't help much here.

Strong agree.

**Comment by wemrysi** [comment]
Closing in favor of #4660.

**Pull request closed without merging**